examples/seismic/model.py

```diff
@@ -722,4 +791,98 @@ def critical_dt(self):
         #
         # The CFL condtion is then given by
         # dt < h / (sqrt(2) * max(vp)))
-        return self.dtype(.5*mmin(self.spacing) / (np.sqrt(2)*mmax(self.vp)))
+        # FIXME: Fix 'Constant' so that that mmax(self.vp) returns the data value
+        return self.dtype(.5*mmin(self.spacing) / (np.sqrt(2)*mmax(self.vp.data)))
+
+
+class ModelViscoelastic(GenericModel):
+    """
+    The physical model used in seismic inversion processes.
+
+    Parameters
+    ----------
+    origin : tuple of floats
+        Origin of the model in m as a tuple in (x,y,z) order.
+    spacing : tuple of floats, optional
+        Grid size in m as a Tuple in (x,y,z) order.
+    shape : tuple of int
+        Number of grid points size in (x,y,z) order.
+    space_order : int
+        Order of the spatial stencil discretisation.
+    vp : float or array
+        P-wave velocity in km/s.
+    qp : float or array
+        P-wave quality factor (dimensionless).
+    vs : float or array
+        S-wave velocity in km/s.
+    qs : float or array
+        S-wave qulaity factor (dimensionless).
+    nbpml : int, optional
+        The number of PML layers for boundary damping.
+    rho : float or array, optional
+        Density in kg/cm^3 (rho=1 for water).
+
+    The `ModelElastic` provides a symbolic data objects for the
+    creation of seismic wave propagation operators:
+
+    damp : Function, optional
+        The damping field for absorbing boundary condition.
+    """
+    def __init__(self, origin, spacing, shape, space_order, vp, qp, vs, qs, rho,
+                 nbpml=20, dtype=np.float32):
+        super(ModelViscoelastic, self).__init__(origin, spacing, shape, space_order,
+                                                nbpml=nbpml, dtype=dtype)
+        # Create dampening field as symbol `damp`
+        self.damp = Function(name="damp", grid=self.grid)
+        initialize_damp(self.damp, self.nbpml, self.spacing, mask=True)
+
+        # Create square slowness of the wave as symbol `m`
+        if isinstance(vp, np.ndarray):
+            self.vp = Function(name="vp", grid=self.grid, space_order=space_order)
+            initialize_function(self.vp, vp, self.nbpml)
+        else:
+            self.vp = Constant(name="vp", value=vp)
+        self._physical_parameters = ('vp',)
```

Comment:
nitpicking: `physical_parameters` should be a list and then finally turned into a tuple and assigned to ``self. `